src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -546,6 +528,47 @@ export default class Pilot {
         return [true, readback];
     }
 
+    /**
+     * Cross a fix at a certain altitude
+     *
+     * @for Pilot
+     * @method cross
+     * @param aircraft {AircraftModel}
+     * @param fixName {string} name of the fix
+     * @param altitude {number} the altitude
+     * @return {array}                [success of operation, readback]
+     */
+    cross(aircraft, fixName, altitude) {
+        if (!NavigationLibrary.hasFixName(fixName)) {
+            return [false, `unknown fix '${fixName}'`];
+        }
+
+        if (!this._fms.hasWaypointName(fixName)) {
+            return [false, `fix '${fixName}' is not on our route`];
+        }
+
+        const response = aircraft.validateNextAltitude(altitude);
+
+        if (response && !response[0]) {
+            return response;
+        }
+
+        const waypoint = this._fms.findWaypoint(fixName);
+
+        waypoint.altitudeMaximum = altitude;
+        waypoint.altitudeMinimum = altitude;
```

Comment:
In Java I would agree but in Javascript? What's wrong with the set-syntax feature? I can barely see a test case for `setMaximumAltitude`. Maybe I am missing something here.

@n8rzz

Reply:
There isn't actually anything wrong with doing it this way, however, we prefer a setter method just so the modification of properties is as consolidated as possible.  The way it's written now is totally valid, but, it makes it harder to track down if there are multiple places setting a property.  By consolidating that setter, we always know where the property is actually being changed.

Reply:
It's mostly because we want to protect the property from messed up values. If there's a setting method which checks the values for validity, then we can avoid some really whacky bugs down the line. I know we're validating the altitude a few lines up, but in general we avoid direct assignments like the plague-- especially when making those changes outside of the class being modified.

Reply:
@erikquinn Have you fixed this one? i.e. can we mark it as resolved or is there something left to do here?

Reply:
The methods have been added. Resolving.